frontend/src/APIClients/AuthAPIClient.ts

```diff
@@ -96,7 +96,7 @@ const register = async (
   lastName: string,
   email: string,
   password: string,
-): Promise<AuthenticatedUser | null> => {
+): Promise<RegisterResponse> => {
   try {
     const { data } = await baseAPIClient.post(
```

Comment:
Later on in this file the local storage token is set (even when it's not supposed to be) - basically this allows the user to bypass the 2FA page. You can verify this if you signup, go to the 2FA page, and refresh. All you need to do is remove line 106 and you're chilling